src/hist/histcustommodes.cpp

```diff
@@ -259,6 +271,106 @@ void HistCustomModes::setInstableAmplitude(double instableAmplitude)
+{ 
+  std::random_device rd;
+  unsigned seed;
+  switch(_seedType){
+    case None : seed = 0;
+      break;
+    case Time : seed = std::chrono::system_clock::now().time_since_epoch().count();
+      break;
+    case Random :
+      seed = rd();
+      break;
+    case User:
+      seed = _seed;
+      break;
+  }
+  std::default_random_engine engine;
+  std::uniform_real_distribution<double> randomDistrib(-dist_max,dist_max);
+  engine.seed(seed);
+    for ( unsigned j = 0 ; j < 3 ; ++j ) {	
+      _strainAmplitudes[j] = randomDistrib(engine);
+    
+  }
+}
+
+geometry::mat3d HistCustomModes::defStrainMatrix(std::vector <double>  Amplitudes)
+{
+  geometry::mat3d strainTot;
+  double epsilon=Amplitudes[0];
+  double delta1 = Amplitudes[1];
+  double delta2 = Amplitudes[2];
+  geometry::mat3d strainTetra = {epsilon * delta1, 0.0 ,0.0,
+                                 0.0 , epsilon * delta1, 0.0,
+                                 0.0 , 0.0 , epsilon * (2*delta1 + delta1*delta1)/((1+delta1)*(1+delta1)) };
+  geometry::mat3d strainShear = {0.0 , delta2 , 0.0,
+                                 delta2 , 0.0 , 0.0 ,
+                                 0.0 , 0.0 , -delta2*delta2/(1-delta2*delta2) };
+  this->rotateStrain(strainTetra);
+  this->rotateStrain(strainShear);
+
+  strainTot[geometry::mat3dind(1,1)] = strainTetra[geometry::mat3dind(1,1)] + strainShear[geometry::mat3dind(1,1)];
+  strainTot[geometry::mat3dind(1,2)] = strainTetra[geometry::mat3dind(1,2)] + strainShear[geometry::mat3dind(1,2)];
+  strainTot[geometry::mat3dind(1,3)] = strainTetra[geometry::mat3dind(1,3)] + strainShear[geometry::mat3dind(1,3)];
+
+  strainTot[geometry::mat3dind(2,1)] = strainTetra[geometry::mat3dind(2,1)] + strainShear[geometry::mat3dind(2,1)];
+  strainTot[geometry::mat3dind(2,2)] = strainTetra[geometry::mat3dind(2,2)] + strainShear[geometry::mat3dind(2,2)];
+  strainTot[geometry::mat3dind(2,3)] = strainTetra[geometry::mat3dind(2,3)] + strainShear[geometry::mat3dind(2,3)];
+
+  strainTot[geometry::mat3dind(3,1)] = strainTetra[geometry::mat3dind(3,1)] + strainShear[geometry::mat3dind(3,1)];
+  strainTot[geometry::mat3dind(3,2)] = strainTetra[geometry::mat3dind(3,2)] + strainShear[geometry::mat3dind(3,2)];
+  strainTot[geometry::mat3dind(3,3)] = strainTetra[geometry::mat3dind(3,3)] + strainShear[geometry::mat3dind(3,3)];
+
+return strainTot;
+}
+
+
+void HistCustomModes::setStrainDir(bool x, bool y, bool z)
+{ 
+  _strainDir.clear();
+  if(x==true)  
+    _strainDir.push_back(StrainDir::x);
+  if(y==true)
+    _strainDir.push_back(StrainDir::y);
+  if(z==true)
+    _strainDir.push_back(StrainDir::z);
+}
+
+void HistCustomModes::rotateStrain(geometry::mat3d &strainMatrix)
+{   
+    using namespace geometry;
+    for ( auto e : _strainDir ) std::cout << e << std::endl;
+    unsigned nchoice = _strainDir.size();
+    std::default_random_engine engine;
+    std::uniform_real_distribution<int> randomDistrib(0,nchoice-1);
```

Comment:
You cannot have a `uniform_real_distribution` of type `int`.
```suggestion
    std::uniform_int_distribution<int> randomDistrib(0,nchoice-1);
```